util/efd_sched.py

```diff
@@ -0,0 +1,231 @@
+        The name of the EFD to connect to (default is 'usdf_efd').
+    time_window_seconds : `float`, optional
+        The size of the time window to search for scheduler URLs, in seconds
+        (default is 2).
+
+    Returns
+    -------
+    scheduler_urls : `pandas.DataFrame`
+        A DataFrame containing the scheduler snapshot times URLs.
+    """
+    desired_time = Time(desired_time)
+    start_time = desired_time - (time_window_seconds / 2) * u.second
+    end_time = desired_time + (time_window_seconds / 2) * u.second
+
+    efd_client = EfdClient(efd)
+    topic = "lsst.sal.Scheduler.logevent_largeFileObjectAvailable"
+    fields = ["url"]
+    scheduler_urls = await efd_client.select_time_series(topic, fields, start_time, end_time)
+    scheduler_urls.index.name = "time"
+    scheduler_urls = scheduler_urls.reset_index()
+    return scheduler_urls
+
+
+async def query_night_schedulers(night, efd="usdf_efd"):
+    """Query the EFD for the night schedulers
+
+    Parameters
+    ----------
+    night : `str`, 'float', or `astropy.time.Time`
+        The night to query, in YYYY-MM-DD format
+    efd : `str`
+        The name of the EFD to query, usdf_efd or summit_efd
+        Defaults to usdf_efd
+
+    Returns
+    -------
+    schedulers : `list`
+        A list of the schedulers for the night
+    """
+
+    try:
+        reference_time = Time(night)
+    except ValueError:
+        if isinstance(night, int):
+            reference_time = Time(night, format="mjd")
+        else:
+            raise ValueError("night must be a valid date in YYYY-MM-DD format")
+
+    # The offset by longitude moves to local solar time.
+    # The offset of 1 makes the night specificed refer to the
+    # local date corresponding to sunset.
+    local_midnight = Time(reference_time.mjd + 1 - Site("LSST").longitude / 360, format="mjd")
+    time_window_seconds = u.day.to(u.second)
+    scheduler_urls = await query_schedulers_in_window(
+        local_midnight, efd=efd, time_window_seconds=time_window_seconds
+    )
+    return scheduler_urls
+
+
+def get_scheduler(scheduler_url, destination=None):
+    """Load a scheduler, conditions tuple from a URI and optionally save it.
+
+    Parameters
+    ----------
+    scheduler_url : `str`
+        The URL of the scheduler to load.
+    destination : `str`, optional
+        The file path to save the scheduler to.
+        If None, the (scheduler, conditions) tuple will be returned.
+
+    Returns
+    -------
+    result : `tuple` or `str`
+        If `destination` is None, the loaded scheduler (scheduler, conditions)
+        tuple is returned.
+        Otherwise, the file path of the saved scheduler is returned.
+
+    Notes
+    -----
+    This function sets the environment variable
+    `LSST_DISABLE_BUCKET_VALIDATION` to "1".
+    """
+
+    os.environ["LSST_DISABLE_BUCKET_VALIDATION"] = "1"
+    scheduler_resource_path = ResourcePath(scheduler_url.replace("https://s3.cp.lsst.org/", "s3://rubin:"))
```

Comment:
I think this access to the LFA needs to be location dependent also. From the summit, you do actually use s3.cp.lsst.org and not s3://rubin. 

Perhaps this argues for setting "location" for both get_scheduler and the methods above, which would be either of "summit" or "usdf"? (summit EFD = "summit_efd", usdf EFD = "usdf_efd", summit LFA = "https://s3.cp.lsst.org/<path>" usdf LFA = "s3://rubin:<path>")  That way it's the same kwarg for all of these methods, and we can just default to usdf (but then a user could just substitute "summit" in all places if at summit). 
In addition, I think it would likely future-proof this a bit if instead of replacing the summit URL explicitly, we split the path (ResourcePath.path() will return the path .. might have to rstrip('/') still, but then add it onto the rest of the url?)

Reply:
This is a bit of an expansion of scope for this issue. I had not intended for this code ever to be run anywhere other than the USDF: it's a development tool, not a production too. It is unlikely ever to be used once PREOPS-4097 is addressed, which I expect to be relatively soon, so "future-proofing" it seems a bit pointless. Still, this seems straightforward, so I'll go ahead and do it.

Reply:
Done.